teos/src/api/internal.rs

```diff
@@ -566,6 +610,30 @@ mod tests_private_api {
                     )) if Locator::from_slice(app_loc).unwrap() == locator
                 ));
             }
+
+            // Query for the current locator with the optional user_id present
+            let response = internal_api
+                .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                    locator: locator.to_vec(),
+                    user_id: Some(random_user_id.to_vec()),
+                }))
+                .await
+                .unwrap()
+                .into_inner();
+
+            // Verify that only a single appointment is returned
+            assert_eq!(response.appointments.len(), 1 );
+
+            // Verify that the appointment have the current locator
+            assert!(matches!(
+                response.appointments[0].appointment_data,
+                Some(common_msgs::appointment_data::AppointmentData::Appointment(
+                    common_msgs::Appointment {
+                        locator: ref app_loc,
+                        ..
+                    }
+                )) if Locator::from_slice(app_loc).unwrap() == locator
+            ));
         }
     }
```

Comment:
tbh, I think this test is now kinda confusing. would go for splitting it into two. leave the current one as is and add a new one that tests the `user_id` filter functionality.

also would use names `our_user_id` & `our_appointments_count`. and maybe `distinct_appointment_numbers` be `other_appointments_count`